apps/prairielearn/src/ee/lib/validateHTML.ts

```diff
@@ -645,5 +646,13 @@ export function validateHTML(file: string, optimistic: boolean, usesServerPy: bo
     errors.push(`Create a server.py file to generate the following: ${templates.join(', ')}`);
   }
 
+  for (const template of templates) {
+    if (!mustacheTemplateChecker.test(template)) {
+      errors.push(
+        `Template of ${template} must be in mustache format. Please adjust so that any logic done in the template is instead done in server.py`,
+      );
+    }
+  }
+
```

Comment:
Can we include some tests, both positive and negative?

Reply:
I started trying to do this when I realized that there were some issues both with the existing and new implementation, which is actually where https://github.com/PrairieLearn/PrairieLearn/pull/11962 came from. I'll add tests once this PR is rebased on that.

Reply:
I added tests in f99e574.

Reply:
Thanks!